src/x3d/opengl/castlescreeneffects.pas

```diff
@@ -696,7 +743,14 @@ procedure TCastleScreenEffects.RenderOverChildren;
       AttribTexCoord.EnableArrayVector2(SizeOf(TScreenPoint),
         OffsetUInt(ScreenPoint[0].TexCoord, ScreenPoint[0]));
 
-      glDrawArrays(GL_TRIANGLE_FAN, 0, 4);
+      if Blending then
```

Comment:
I realized that the condition for this should be `if Blending and "this is the last effect, rendered on screen" then` ...

I.e. the intermediate effects, that reuse FBOs, should not use blending (instead, they should rely on `TCastleSimpleBackground` clearning the contents).

Please extend the `RenderOneEffect` routine with additional parameter `const RenderToScreen: Boolean`.

Then it should be used with `false` in the loop, `RenderOneEffect(GetScreenEffect(I), false);`

And used with `true` for final render, `RenderOneEffect(GetScreenEffect(CurrentScreenEffectsCount - 1), true);`

Reply:
Done in bebc4079783d87ff83f3cfd25bd46aabd7506a1d !